### PR TITLE
Issue #8979: fix for 'array initialization rcurly' has incorrect indentation

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
Fixes Issue #8979 

Diff Regression projects: https://gist.githubusercontent.com/Abhishek-kumar09/0745ecf11934e0214b9187f9d4896e4a/raw/7994fb09f0b5d073f1937e62faecdd5cb3908e64/project-to-test-on.properties
Diff Regression config: https://gist.githubusercontent.com/Abhishek-kumar09/0745ecf11934e0214b9187f9d4896e4a/raw/7e9f169832e45c826fb99e72b3ff08f32765243a/my_check.xml

### CODE REVIEW

**Comment by Abhishek-kumar09** [comment]
Github, generate report

**Comment by romani** [comment]
Github, rebase

**Comment by romani** [comment]
Github, rebase

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationValidArrayInitIndentTwo.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/* Config:                                                                  //indent:0 exp:0
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationValidArrayInitIndentTwo.java:1]: please add file with invalid Array indentation.

**Review by romani — CHANGES_REQUESTED** [review]
please make some Inputs with multidimensional multi-line arrays (valid and not valid)
insspired by code [here](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/array_rcurly_2020185028/reports/diff/guava/index.html#A87)
```
61          new Object[][] {
62            {UNDIRECTED,
                      value2
                   }, 
               {DIRECTED},
63          });
```

item:

**Comment by Abhishek-kumar09** [comment]
The children of 2d array is not following arrayInitIndent, I think I have to work on it.
```
class T {
  int[][] a =  {
    new int[] {1, 2},
    {1, 2},
  }
}

```
error: 
```
Starting audit...
[ERROR] /home/abhishhek/playground/t5.java:4:5: 'array initialization lcurly' has incorrect indentation level 4, expected level should be one of the following: 6, 8, 12. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationValidArrayInitIndentTwo.java (hunk 1 above)
Comment (Abhishek-kumar09) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationValidArrayInitIndentTwo.java:1]: @romani Done.

**Comment by romani** [comment]
Github, rebase

**Comment by Abhishek-kumar09** [comment]
Github, rebase

**Comment by romani** [comment]
Github, rebase

**Review by romani — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ArrayInitHandler.java [hunk 2]
```diff
@@ -74,7 +75,16 @@ protected DetailAST getLeftCurly() {
...
     @Override
     protected IndentLevel curlyIndent() {
-        final IndentLevel level = new IndentLevel(getIndent(), getBraceAdjustment());
+        int offset = 0;
+
+        final DetailAST lcurly = getLeftCurly();
+
+        if (isOnStartOfLine(lcurly)
+            && !TokenUtil.isOfType(lcurly.getParent(), TokenTypes.ARRAY_INIT)) {
+            offset = getBraceAdjustment();
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ArrayInitHandler.java:84]: `curlyIndent` is returning the expected indentation for the left and right curlies.

If the parent of ARRAY_INIT isn't an ARRAY_INIT (which is almost a non-brace, then how can we use brace adjustment here which is only to be used for braces. Shouldn't `basicOffset` or `lineWrappingIndentation` be used here instead?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ArrayInitHandler.java [hunk 3]
```diff
@@ -74,7 +75,16 @@ protected DetailAST getLeftCurly() {
 
     @Override
     protected IndentLevel curlyIndent() {
-        final IndentLevel level = new IndentLevel(getIndent(), getBraceAdjustment());
+        int offset = 0;
+
+        final DetailAST lcurly = getLeftCurly();
+
+        if (isOnStartOfLine(lcurly)
+            && !TokenUtil.isOfType(lcurly.getParent(), TokenTypes.ARRAY_INIT)) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ArrayInitHandler.java:83]: If you are just comparing one type, then why don't you just use an equal sign? I am not seeing a need to use the util method here.

Comment (Abhishek-kumar09) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ArrayInitHandler.java]: Corrected

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ArrayInitHandler.java (hunk 2 above)
Comment (Abhishek-kumar09) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/ArrayInitHandler.java:83]: @rnveach, sorry for late reply, I got disconnected with my own code.
 
It is for the case when lcurly is on new line, then curlies should be indented on `braceAdjustment` value ahead of the base.
It is for old style java codes.
```
class A 
  { // braceAdj
      void main() {} // braceAdj + basicOffset
  }  
```
```
class A {
    void main() {} //  basicOffset
}  
```

doc: 
image

**Comment by romani** [comment]
@rnveach , please finish review

**Comment by romani** [comment]
@rnveach , ping.

**Comment by romani** [comment]
Github, rebase

**Review by rnveach — APPROVED** [review]

**Pull request merged**